src/SSHDebugPS/Docker/DockerHelper.cs

```diff
@@ -17,6 +18,107 @@ public class DockerHelper
         private const string dockerPSCommand = "ps";
         // --no-trunc avoids parameter truncation
         private const string dockerPSArgs = "-f status=running --no-trunc --format \"{{json .}}\"";
+        public const string dockerVersionCommand = "version";
+        public const string dockerVersionArgs = "--format {{.Server.Os}}";
+        public const string dockerInfoCommand = "info";
+        public const string dockerInfoArgs = "--format {{.Driver}}";
+
+        public static string GetDockerOutputString(string hostname, string dockerCommand, string dockerArgs)
+        {
+            string path = null;
+            try
+            {
+                ThreadHelper.JoinableTaskFactory.Run(async () =>
+                {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    path = GetDockerOutputStringImpl(hostname, dockerCommand, dockerArgs);
+                });
+            }
+            catch (CommandFailedException ex)
+            {
+                string errorMessage = UIResources.CommandExecutionErrorFormat.FormatCurrentCultureWithArgs(dockerCommand, ex.Message);
+                throw new CommandFailedException(errorMessage, ex);
```

Comment:
Why catch and throw a new one? can you format the error where the original throw is and then just do `throw` here to rethrow? throwing new exceptions is expensive and this seems to be for control flow.